mutation_writer/token_group_based_splitting_writer.hh

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (C) 2023-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#pragma once
+
+#include <functional>
+
+#include "feed_writers.hh"
+
+namespace mutation_writer {
+
+using token_group_id = size_t;
+using classify_by_token_group = std::function<token_group_id(dht::token)>;
```

Comment:
Do we expect to have more than two groups here? If not, we can just pass a single token and have the segregator push all tokens `<` to the left and all `>=` to the right buckets.

Reply:
I expect @raphaelsc is pushing for 1:2^n rather than 1:2 splits in the future. If we see a table is growing rapidly, we want to split quickly as long as nr_tablets < nr_shards_in_rack so it can use all cpu.